python/deltalake/table.py

```diff
@@ -747,6 +781,309 @@ def delete(self, predicate: Optional[str] = None) -> Dict[str, Any]:
         return json.loads(metrics)
 
 
+class TableMerger:
+    """API for various table MERGE commands."""
+
+    def __init__(
+        self,
+        table: DeltaTable,
+        source: Union[pyarrow.Table, pyarrow.RecordBatch],
+        predicate: str,
+        source_alias: Optional[str] = None,
+        target_alias: Optional[str] = None,
+        safe_cast: bool = True,
+    ):
+        self.table = table
+        self.source = source
+        self.predicate = predicate
+        self.source_alias = source_alias
+        self.target_alias = target_alias
+        self.safe_cast = safe_cast
+        self.writer_properties: Optional[Dict[str, Optional[int]]] = None
+        self.matched_update_updates: Optional[Dict[str, str]] = None
+        self.matched_update_predicate: Optional[str] = None
```

Comment:
I went off the examples in delta-spark, but we can apparently do multiple clauses in the Rust API. I think instead of passing it a value, can pass perhaps a list of dicts, having this structure:` { predicate: "x > 1", updates: {"col" : "col"}}`
And then add an iterator for each of those in rust. 

Should we get it in this PR or maybe I can add it in a smaller PR updating the logic afterwards?

Reply:
I'm fine making that a follow up, but thought I would point that out.

Reply:
Yes good for pointing it out! I will do that in a follow-up then.